src/main/scala/SIFAstExtensions.scala

```diff
@@ -125,6 +125,21 @@ case class SIFLowExp(exp: Exp, comparator: Option[String] = None, typVarMap: Map
   override val extensionIsPure: Boolean = exp.isPure
 }
 
+case class SIFRelExp(exp: Exp, i: IntLit)

```

Comment:
Are you using `IntLit` instead of `Int` so that Viper's rewriter is not breaking? Does it make sense to document this? Also, you should comment which index corresponds to which execution.